MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -18,17 +18,30 @@ final class FeedViewModel {
 		Localized.Feed.title
 	}
 	
+    var errorMessage: String {
+        Localized.Feed.loadError
+    }
+
+    //MARK: - Observers
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
-	
+    var onError: Observer<String>?
```

`onError: Observer<String> ` could be `onErrorStateChange: Observer<String?>`, so the ViewModel decides when to show/hide errors (instead of the view controller). It's often easier to manage state when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed.